src/test/fuzz.cpp

```diff
@@ -1,144 +1,83 @@
 
-#include "test/fuzz.h"
-
-#include <signal.h>
+#include <algorithm>
+#include <autocheck/autocheck.hpp>
 #include <xdrpp/autocheck.h>
-#include <xdrpp/printer.h>
-
 /**
  * This is a very simple fuzzer _stub_. It's intended to be run under an
- * external fuzzer with some fuzzing brains, AFL or fuzzgrind or whatever.
+ * external fuzzer with some fuzzing brains, at this time, preferably AFL.
  *
  * It has two modes:
  *
- *   - In gen-fuzz mode it spits out a small file containing a handful of
- *     random StellarMessages. This is the mode you use to generate seed data
- *     for the external fuzzer's corpus.
+ *   - In genfuzz mode it spits out a small file containing a handful of
+ *     random FuzzTransactionInputs or StellarMessages. This is the mode you use
+ *     to generate seed data for the external fuzzer's corpus.
  *
- *   - In fuzz mode it reads back a file and appplies it to a pair of of
- *     stellar-cores in loopback mode, cranking the I/O loop to simulate
- *     receiving the messages one by one. It exits when it's read the input.
- *     This is the mode the external fuzzer will run its mutant inputs through.
+ *   - In fuzz mode it reads back a file and appplies it to a stellar-core
+ *     instance, applying but not committing transactions one by one to simulate
+ *     certain transaction/overlay scenarios. It exits when it's applied the
+ *     input. This is the mode the external fuzzer will run its mutant inputs
+ *     through.
  *
  */
 
-namespace stellar
-{
+using namespace stellar;
 
-std::string
-msgSummary(StellarMessage const& m)
-{
-    xdr::detail::Printer p(0);
-    xdr::archive(p, m.type(), nullptr);
-    return p.buf_.str() + ":" + hexAbbrev(sha256(xdr::xdr_to_msg(m)));
-}
-
-bool
-tryRead(XDRInputFileStream& in, StellarMessage& m)
+template <typename XDR_TYPE, typename Predicate>
+void
+fuzzInputGenerator(const std::string& filename, Predicate isDesiredXDR)
 {
+    XDROutputFileStream out;
+    out.open(filename);
+    autocheck::generator<XDR_TYPE> gen;
     try
     {
-        return in.readOne(m);
+        XDR_TYPE m(gen(10));
+        while (!isDesiredXDR(m))
+            m = gen(10);
+
+        out.writeOne(m);
     }
-    catch (xdr::xdr_runtime_error& e)
+    catch (xdr::xdr_bad_discriminant const&)
     {
-        LOG(INFO) << "Caught XDR error '" << e.what()
-                  << "' on input substituting HELLO";
-        m.type(HELLO);
-        return true;
     }
 }
 
-void
-seedRandomness()
+namespace stellar
+{
+
+namespace fuzztest
+{
+std::unique_ptr<Fuzzer>
+createFuzzer(int processID, FuzzerMode fuzzerMode)
 {
-    srand(1);
-    gRandomEngine.seed(1);
+    switch (fuzzerMode)
+    {
+    case FuzzerMode::OVERLAY:
+        return std::make_unique<OverlayFuzzer>();
+    case FuzzerMode::TRANSACTION:
+    {
+    }
+    }
+}
 }
 
-#define PERSIST_MAX 10000000
-#define INITIATOR 1
-#define ACCEPTOR 0
+#define PERSIST_MAX 1000000
+#define MAX_XDR_SIZE 0x1000000
```

Comment:
I don't think you should use this value `0x1000000` for transactions (way too big)

Reply:
Ah yes... was just reading how someone 8x'd their speed by limiting to 50kb

Reply:
Averaged XDR size of 100k `path_payment` only XDR's in the range of 1-3 ops and got `294 bytes`. So I think `0x500` should be fine.